pyccel/parser/scope.py

```diff
@@ -476,7 +476,7 @@ def insert_symbol(self, symbol):
                 self.parent_scope.insert_symbol(symbol)
             elif symbol not in self._used_symbols:
                 collisionless_symbol = self.name_clash_checker.get_collisionless_name(symbol,
-                        self._used_symbols.values())
+                        self.all_used_symbols)
```

Comment:
The properties `local_used_symbols` and `all_used_symbols` are very useful and should be mentioned in the developers docs, if they are not already!

By the way, it appears that `local_used_symbols` returns a `dict` while `all_used_symbols` returns a `set`. What is the reason for this choice?

Reply:
`local_used_symbols` gets the `_used_symbols` object which is a dictionary. It is used for initialising new scopes, usually for initialising the semantic equivalent of a syntactic scope.

`all_used_symbols` gets all the symbols which are currently used in the scope (including enclosing scopes). It is almost exclusively used for obtaining new names while avoiding collisions with existing names.

These are not currently mentioned in the developers docs as I didn't have a place in mind where developers frequently need these objects. These docs are supposed to supplement the docstrings not replace them entirely so I would hope that devs would go looking for functions if they need them. I can add something though

Reply:
[4f10089](https://github.com/pyccel/pyccel/pull/1652/commits/4f100894c7ff46b4a5a7c75a670c2f3c8322a798)